### ISSUE TITLE
Bug: permission_handler doesn't support iOS while checking these permissions.

### ISSUE THREAD

**Issue by 0xharkirat** [CCExtractor/ultimate_alarm_clock#497]
### This function:
```
 checkOverlayPermissionAndNavigate() async {
    if (!(await Permission.systemAlertWindow.isGranted) ||
        !(await Permission.ignoreBatteryOptimizations.isGranted) ||
        !(await Permission.notification.isGranted)) {
        ...other code
}
```
 <hr>

- code asking systemAlertWindow permission:
```
              // Request overlay permission
              if (!(await Permission.systemAlertWindow.isGranted)) {
                final status = await Permission.systemAlertWindow.request();
                if (!status.isGranted) {
                  debugPrint('SYSTEM_ALERT_WINDOW permission denied!');
                  return;
                }
              }
```
_output:_
```
flutter: SYSTEM_ALERT_WINDOW permission denied!
```
<hr>

- code asking ignoreBatteryOptimizations permissions:
```
              if (!(await Permission.ignoreBatteryOptimizations.isGranted)) {
                bool requested = await Permission.ignoreBatteryOptimizations
                    .request()
                    .isGranted;
                if (!requested) {
                  debugPrint('IGNORE_BATTERY_OPTIMIZATION permission denied!');
                  return;
                }
              }
```

_Output:_
```
flutter: IGNORE_BATTERY_OPTIMIZATION permission denied!
```
<hr>
  
### [permission_handler](https://pub.dev/packages/permission_handler) plugin [code](https://github.com/Baseflow/flutter-permission-handler/blob/main/permission_handler_platform_interface/lib/src/permissions.dart) (see comments, Android only):

- For system alert window:

```
 /// Permission for creating system alert window (Android only).
  ///
  /// Allows an app to create windows shown on top of all other apps.
  static const systemAlertWindow = Permission._(23);
```

- For ignore battery optimisation:
```
 /// Permission for accessing ignore battery optimizations (Android only).
  static const ignoreBatteryOptimizations = Permission._(16);
```

**Comment by 0xharkirat** [comment]
This bug is causing UI Navigation problems like this:

1. When we press update button, the permission dialog box appears.
2. if we press grant permissions button, it sets the alarm, closes the dialog box, prints `flutter: SYSTEM_ALERT_WINDOW permission denied!` **but** does not navigate back to the home screen.
3. User is confused whether the alarm is set or not.
4. But when user goes back manually, alarm is set.

**Comment by MarkisDev** [comment]
Merged with #499
